### DISCUSSION TITLE
How to compute the modes information of CMS system after assembling?

### DISCUSSION THREAD

**Discussion by qiushi-Lee** [jgerstmayr/EXUDYN#23] [Q&A on usage of Exudyn] [answered]
I would like to assemble a whole part by some ObjectFFRFreducedOrderInterface() subparts and compute the corresponding modes information, such as eigen frequency and eigenmodes. However, I find that the functions to compute modes information, such as ComputeEigenmodes() and ComputeEigenmodesNGsolve() are all subject to the FEM CLASS. They can not output the modes information of a assembled CMS system. 
Just like ANSYS and other commersial CAE software, they can divide a part into some subparts and compute these modes. Then, Some boundary conditons are inserted into the interfaces, and the whole part is assembled by the modes of subparts via CMS method. Finally, it can output the modes information of the whole part.
Does Exudyn have the corresponding function to solve the question just as above statement?

**Comment by jgerstmayr** [comment, marked as answer]
The pre-processing is just available for single parts. However, if you assemble a system in an MBS, there is the possibility to perform "ComputeEigenModes". This, however, requires that the subparts are rigidly connected. But I think, would you need is to assemble parts by merging meshes, which is a functionality for meshers or FE-Tools.

**Reply by qiushi-Lee** [reply]
Thanks for your reply. As far as I know, system modes analysis has two methods. One is assembling the system at meshing step and directly compute the modes information through the whole system mesh. The other is CMS method. Firstly, getting the main modesbasis and constraint modes of subparts, then, assembing them to compute modes information. I would like to use EXUDYN to achieve the latter method. For example, I divide a block into three parts along lengthwise direction. Two parts are reduced via HCB method, then assembling them to compute modes information and show the visualization of system modes.
In order to finish above example, I meet two main problems. 

1. The connecting problem between two parts, which have different nodes on the interfaces. As your statement in Doc document, HCB method remain six constraint modes to represent boundary nodes set. Does that means the mean position of the interface nodes need to  align with the other connecting interface.
2. The AnimateModes() function seem like only to show the modes of one single object of MBS system.

Hope to get your further suggestion. I am really appreciate your kind answer for the beginner of EXUDYN.

**Reply by jgerstmayr** [reply]
1. the interface nodes of two bodies do not need to align; you could connect them with a GenericJoint; to perform further system reduction, you have to extend the Python code
2. AnimateModes() only visualizes one object by putting a sinusoidal change on one coordinate.
